Indoor_GPS/marvelmind.py

```diff
@@ -108,14 +110,24 @@ def print_position(self):
             print ("Hedge {:d}: X: {:.3f}, Y: {:.3f}, Z: {:.3f}, Angle: {:d} at time T: {:.2f}".format(self.position()[0], self.position()[1], self.position()[2], self.position()[3], self.position()[4], self.position()[5]/1000.0))
 
     def position(self):
-        return list(self.valuesUltrasoundPosition)[-1];
+        ans = list(self.valuesUltrasoundPosition)[-1]
+        if self.num_called < 5:
+            if self.num_called == 0:
+                self.zero_pos = np.array(ans)
+            else:
+                self.zero_pos = ((self.num_called) * self.zero_pos + np.array(ans))/(self.num_called + 1)
```

Comment:
I don't really understand this part. Could you explain a bit?

Reply:
This method was re-written so that its return value is the vector from zero_pos to the current position of the hedge. The variable `num_called` stores the number of times this method has returned a value so far.

During the first k <= 5 calls, we update the `zero_pos` attribute to be the average of the first k positions: e.g. the average of the first 4 positions is equal to (sum of first 4 pos) / 4 = ((3 * avg of first 3 pos) + 4th pos) / (3 + 1).

After the 5th call, `zero_pos` isn't modified anymore.